src/widgets/chart.rs

```diff
@@ -102,11 +102,14 @@ pub struct Dataset<'a> {
 }
 
 impl<'a> Dataset<'a> {
+    /// Assigns a name to this dataset, making it appear on the [Chart] legend.
+    /// Datasets don't require a name and can be created without specifying one.
+    /// Once assigned, a name can't be removed, only changed.
```

Comment:
Nit: doc comments generally have a header and body like:

```suggestion
    /// Assigns a name to this dataset
    ///
    /// Named datasets will appear in the [Chart] legend. This is an optional property, Datasets
    /// don't require a name and can be created without specifying one.
```